### PR TITLE
Integration tests for SCIM permissions API

### PR DESCRIPTION
**Pull request by GDRDABARERA**
Depends on https://github.com/wso2/carbon-identity-framework/pull/2413 and https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/199

Tests for https://github.com/wso2/product-is/issues/6591

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1841,7 +1841,7 @@
 
         <!--Carbon Identity Framework Version-->
 
-        <carbon.identity.framework.version>5.14.62</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.14.66-SNAPSHOT</carbon.identity.framework.version>
```
Comment (GDRDABARERA) [inline pom.xml]: Need to change once dependent PRs merged

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            swaggerDefinition = getAPISwaggerDefinition(FILE_BASE_PATH + swaggerFilePath);
+        } catch (IOException e) {
+            Assert.fail(String.format("Unable to read the swagger definition from %s", API_DEFINITION_FILE_NAME), e);
+        }
+    }
+
+    public void testInit( String apiDefinition, String tenantDomain)
+            throws AxisFault {
+
+        String basePath;
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java]: Make this final. It makes sure you only instantiate this variable once.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        String basePathInSwagger = SCIM2_BASE_PATH_IN_SWAGGER;
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java]: remove this and use the class level static constant

File: modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-patch-remove-group-permissions.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "Operations": [
+    {
+      "op": "remove",
+      "value":
+      [
+        "/permission/admin/configure/security/usermgt"
+      ]
+    }
+  ]
+}
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-patch-remove-group-permissions.json]: Missing new line

File: modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-patch-add-group-permissions.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  "Operations": [
+    {
+      "op": "add",
+      "value":
+      [
+        "/permission/admin/manage/search"
+      ]
+    }
+  ]
+}
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-patch-add-group-permissions.json]: Missing new line

File: modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-add-group.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "displayName":"manager"
+}
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/resources/org/wso2/identity/integration/test/scim2/rest/api/scim2-add-group.json]: Missing new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        String responseString = response.then()
+                                        .log().ifValidationFails()
+                                        .assertThat()
+                                        .statusCode(HttpStatus.SC_OK)
+                                        .extract()
+                                        .asString();
+        Assert.assertNotNull(responseString);
+        // PUT operation should return the same response as the requested.
+        Assert.assertEquals(responseString, body, "The response for the PUT operation is incorrect.");
+
```
Comment (Kavindu-Dodan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java]: code formatting - unwatned spaces ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+    private static final String API_DEFINITION_FILE_NAME = "scim2.yaml";
+
+    protected static String swaggerDefinition;
+    protected static final String SCIM2_BASE_PATH_IN_SWAGGER = "/t/\\{tenant-domain\\}" + SCIM2_ENDPOINT;
+    protected static final String SCIM2_BASE_PATH_WITH_TENANT_CONTEXT = TENANT_CONTEXT_IN_URL + SCIM2_ENDPOINT;
+    public static final String FILE_BASE_PATH = (System.getProperty("basedir", "."))
+            + File.separator + "src" + File.separator + "test"
+            + File.separator + "resources" + File.separator;
+    private static final String basePathInSwagger = SCIM2_BASE_PATH_IN_SWAGGER;
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2BaseTest.java]: why do we need another constant? Can't we use SCIM2_BASE_PATH_IN_SWAGGER?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.GROUPS_ENDPOINT;
+import static org.wso2.identity.integration.test.scim2.SCIM2BaseTestCase.PERMISSIONS_ENDPOINT;
+
+public class SCIM2GroupTest extends SCIM2BaseTest {
+
+    private static final Log log = LogFactory.getLog(SCIM2GroupTest.class);
+
+    protected String END_POINT_URL;
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java]: This format is used for constants. If this is not a constant, make it camel case.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+public class SCIM2GroupTest extends SCIM2BaseTest {
+
+    private static final Log log = LogFactory.getLog(SCIM2GroupTest.class);
+
+    protected String endpointURL;
+    protected String groupId = null;
+    private static final String SCIM_CONTENT_TYPE = "application/scim+json";
+    private static final String USER_MGT_PERMISSION = "permission/admin/configure/security/usermgt";
+
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/SCIM2GroupTest.java]: Unnecessary new line.

**Pull request merged**